### DISCUSSION TITLE
Show tides in meters

### DISCUSSION THREAD

**Discussion by JonFear** [Daveiano/weewx-wdc#174] [General]
Hi Dave

Many thanks for v3.2.0. I have only just seen it and so will get it deployed as soon as possible. Switched on notifications correctly now!

Hopefully a simple one, I would like to display the tide heights in meters. Someone else has done this in the forecast skin tide page but unfortunately it does not work in the tide-forecast include page.

Is this simple to implement?

BW

Jon

Code:

```
#if $varExists('forecast')

  <h1>Tide Times for <your location></h1>
  <p>
  <table>
    <tr>
      <td>Date and Time (local) </td><td>Tide H/L </td><td>Height (metres)</td> <— optional header row
    </tr>
    #set $tides = $forecast.xtides
    #if len($tides) > 0
      #for $tide in $tides  
        #set $height=$tide.offset.raw / 3.281 <— add this line here
        <tr>
          <td>$tide.event_ts.format("%a %d %H:%M")</td>
          <td>$tide.hilo</td>
          ## <td> $tide.offset.nolabel('%.1f')</td> <— comment out (##) or remove this line
         <td>
          #echo '%.2f' % $height <— add the converted tide height here
          m <— to show 'm' for metres make sure you add a line break in the table data cell before the m
         </td> 
       </tr>
     #end for
  </table>
#else
  Tide forecast is not available.
#end if
  </p>
#else

  <p>The forecast search list extension is not installed.</p>

#end if
```

**Comment by Daveiano** [comment]
Hi Jon,

> Many thanks for v3.2.0. I have only just seen it and so will get it deployed as soon as possible. Switched on notifications correctly now!

There are a few bugs in 3.2.0 you should be aware of, most to mention a bug with the date/time display on the new sensor status page (see this issue GH-172 and this commit https://github.com/Daveiano/weewx-wdc/commit/178e6ba2de742f6a3adc3b1e3652109b43a29ef7)

Otherwise, I am happy that you are testing it out, and I would be happy to get any feedback ;) 

For the tide display: I wasn't aware that tides are shown in feet, the observation `offset` should be a member of `group_altitude` but it does not to be calculated correctly, but I have no clue yet, why this happens. Yes, your fix would be easy to implement. I created an updated `forecast-table.inc` file in a new branch. 

Diff: https://github.com/Daveiano/weewx-wdc/commit/8191a4b4579dbab2af4df8ee0575fa2f3d59767d  
File for download: https://github.com/Daveiano/weewx-wdc/blob/174-show-tides-in-meters/skins/weewx-wdc/includes/forecast-table.inc

It should work, feel free to test it out!

I will have a look at why this does not get displayed in the correct unit because this should not be statically tied to a unit.

**Comment by JonFear** [comment]
Hi Dave

Sorry no joy! Even with 3.2.0 now installed. see wslwx.co.uk.

BW

Jon

`May 31 07:55:32 weewx weewx[137406] ERROR weewx.cheetahgenerator: Parse error while evaluating file /etc/weewx/skins/weewx-wdc/index.html.tmpl
May 31 07:55:32 weewx weewx[137406] ERROR weewx.cheetahgenerator: **** Ignoring template /etc/weewx/skins/weewx-wdc/index.html.tmplMay 31 07:55:32 weewx weewx[137406] ERROR weewx.cheetahgenerator: **** Reason: #012#012Invalid Syntax#012Line 1989, column 79 in file /etc/weewx/skins/weewx-wdc/includes/forecast-table.inc#012#012Line|Cheetah Code#012----|-------------------------------------------------------------#0121986|        </tr>#0121987|        <tr>#0121988|          <td id="L102" class="blob-num js-line-number js-code-nav-line-number js-blob-rnum" data-line-number="102"></td>#0121989|          <td id="LC102" class="blob-code blob-code-inner js-file-line">      $<span class=pl-s1>filtered_periods</span>.<span class=pl-en>append</span>($<span class=pl-s1>p</span>)</td>#012                                                                                   ^#0121990|        </tr>#0121991|        <tr>#0121992|          <td id="L103" class="blob-num js-line-number js-code-nav-line-number js-blob-rnum" data-line-number="103"></td>  `

**Reply by Daveiano** [reply]
You got a Syntax Error in the template somewhere. Could you post the whole `forecast-table.inc` file here? I am sure it's just a small thing.

**Comment by JonFear** [comment]
Hi Dave

That was just the copy of the forecast-table.inc from the link above...

I did not play with that one at all...

Jon

**Reply by Daveiano** [reply]
Sorry, but this can't be true. In the error message you posted above there is markup which is not included in the source code of the skin like:

`<span class=pl-s1>filtered_periods</span>`

or 

`<td id="L103" class="blob-num js-line-number js-code-nav-line-number js-blob-rnum" data-line-number="103">`

So there is definitely something custom here.

**Comment by JonFear** [comment]
Hi Dave

Weird. OK I will download again and try it. Most strange....

Jon

**Comment by JonFear** [comment]
Hi Dave

These are the steps just taken to get the same result.... even weirder!

Jon

jon@weewx:/etc/weewx/skins/weewx-wdc/includes$ sudo wget https://github.com/Daveiano/weewx-wdc/blob/174-show-tides-in-meters/skins/weewx-wdc/includes/forecast-table.inc
--2023-05-31 12:01:33--  https://github.com/Daveiano/weewx-wdc/blob/174-show-tides-in-meters/skins/weewx-wdc/includes/forecast-table.inc
Resolving github.com (github.com)... 140.82.121.3
Connecting to github.com (github.com)|140.82.121.3|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [text/html]
Saving to: ‘forecast-table.inc.2’
forecast-table.inc.2                  [ <=>                                                       ] 442.88K  2.73MB/s    in 0.2s    
2023-05-31 12:01:33 (2.73 MB/s) - ‘forecast-table.inc.2’ saved [453508]
jon@weewx:/etc/weewx/skins/weewx-wdc/includes$ mv forecast-table.inc forecast-table.inc.3
jon@weewx:/etc/weewx/skins/weewx-wdc/includes$ sudo mv forecast-table.inc forecast-table.inc.3
jon@weewx:/etc/weewx/skins/weewx-wdc/includes$ sudo mv forecast-table.inc.2 forecast-table.inc
jon@weewx:/etc/weewx/skins/weewx-wdc/includes$ sudo systemctl restart weewx

sudo tail -f /var/log/syslog
May 31 12:04:19 weewx weewx[411331] ERROR weewx.cheetahgenerator: Parse error while evaluating file /etc/weewx/skins/weewx-wdc/index.html.tmpl
May 31 12:04:19 weewx weewx[411331] ERROR weewx.cheetahgenerator: **** Ignoring template /etc/weewx/skins/weewx-wdc/index.html.tmpl  
May 31 12:04:19 weewx weewx[411331] ERROR weewx.cheetahgenerator: **** Reason: #012#012Invalid Syntax#012Line 1989, column 79 in file /etc/weewx/skins/weewx-wdc/includes/forecast-table.inc#012#012Line|Cheetah Code#012----|-------------------------------------------------------------#0121986|        </tr>#0121987|        <tr>#0121988|          <td id="L102" class="blob-num js-line-number js-code-nav-line-number js-blob-rnum" data-line-number="102"></td>#0121989|          <td id="LC102" class="blob-code blob-code-inner js-file-line">      $<span class=pl-s1>filtered_periods</span>.<span class=pl-en>append</span>($<span class=pl-s1>p</span>)</td>#012
                                                                         ^#0121990|        </tr>#0121991|        <tr>#0121992|       
   <td id="L103" class="blob-num js-line-number js-code-nav-line-number js-blob-rnum" data-line-number="103"></td>

jon@weewx:/etc/weewx/skins/weewx-wdc/includes$ sudo mv forecast-table.inc forecast-table.inc.2
jon@weewx:/etc/weewx/skins/weewx-wdc/includes$ sudo mv forecast-table.inc.3 forecast-table.inc
jon@weewx:/etc/weewx/skins/weewx-wdc/includes$ sudo systemctl restart weewx

sudo tail -f /var/log/syslog
May 31 12:21:16 weewx weewx[414754] INFO weewx.cheetahgenerator: Generated 12 files for report forecast in 0.21 seconds
May 31 12:21:16 weewx weewx[414754] INFO weewx.reportengine: Copied 0 files to /var/www/html/weewx/forecast
May 31 12:21:16 weewx weewx[414754] DEBUG weewx.reportengine: Running report 'WdcReport'
May 31 12:21:16 weewx weewx[414754] DEBUG weewx.reportengine: Found configuration file /etc/weewx/skins/weewx-wdc/skin.conf for report 'WdcReport'
May 31 12:21:16 weewx weewx[414754] DEBUG weewx.cheetahgenerator: Using search list ['user.weewx_wdc.WdcGeneralUtil', 'user.weewx_wdc.WdcStatsUtil', 'user.weewx_wdc.WdcDiagramUtil', 'user.weewx_wdc.WdcCelestialUtil', 'user.weewx_wdc.WdcArchiveUtil', 'user.weewx_wdc.WdcTableUtil', 'user.weewx_wdc.WdcForecastUtil', 'weewx.cheetahgenerator.Almanac', 'weewx.cheetahgenerator.Current', 'weewx.cheetahgenerator.DisplayOptions', 'weewx.cheetahgenerator.Extras', 'weewx.cheetahgenerator.Gettext', 'weewx.cheetahgenerator.JSONHelpers', 'weewx.cheetahgenerator.PlotInfo', 'weewx.cheetahgenerator.SkinInfo', 'weewx.cheetahgenerator.Station', 'weewx.cheetahgenerator.Stats', 
'weewx.cheetahgenerator.UnitInfo']

**Reply by Daveiano** [reply]
Oh sorry wrong link :D https://github.com/Daveiano/weewx-wdc/blob/174-show-tides-in-meters/skins/weewx-wdc/includes/forecast-table.inc also includes the Github page elements.

Please use

`wget https://raw.githubusercontent.com/Daveiano/weewx-wdc/174-show-tides-in-meters/skins/weewx-wdc/includes/forecast-table.inc`

**Comment by JonFear** [comment]
Ah! That looks better! Now just got to change the text to read meters, that I can do!

Thank you

Jon

**Comment by JonFear** [comment]
Hmm Perhaps I can't change that text so easily! Looks like it is calling from somewhere else.
 <bx-structured-list-cell>
          $gettext("Tides")<br />
          in $unit.label.altitude
        </bx-structured-list-cell>

Jon

**Comment by Daveiano** [comment]
You can just replace `$unit.label.altitude` with `m` or `meter` or whatever you want. 

Note: You need to do this twice: One  time for the daily and one time for the hourly forecast. Here: https://github.com/Daveiano/weewx-wdc/blob/174-show-tides-in-meters/skins/weewx-wdc/includes/forecast-table.inc#L341 and here: https://github.com/Daveiano/weewx-wdc/blob/174-show-tides-in-meters/skins/weewx-wdc/includes/forecast-table.inc#L676

BTW: Which unit_system are you using for your reports? US or metric? 

Read more here: https://weewx.com/docs/customizing.htm#how_to_change_units

**Comment by Daveiano** [comment]
Hi Jon,

I just tested around a bit with the tides and noticed that everything is correctly displayed.

If you are seeing tides in feet, then it's because of your WeeWX configuration. The observation is called `offset` and is a member of the `group_altitude` (https://www.weewx.com/docs/customizing.htm#units). Altitudes can be displayed as feet or meters. The default for the `US` unit system is foot, for the `metric` it's meter. 

On your page, I can see that you are using feet for `group_altitude` because `cloudbase` also shows up in feet. Because of that `offset` also shows as feet.

So you could either switch to meter for the whole `group_altitude` or use the "hack" from above.

This was just a quick summary of what I have found out

**Comment by JonFear** [comment]
Hi Dave

Ah yes! The great British habit of mixing units! Unfortunately Weewx itself does not cope with us Brit's and their passion for the mixed unit schema. Never mind, the hack done works until I can come up with something more friendly. It just takes a little longer to deploy a new version...

Thank you for looking and your advice...

Jon
